src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/InstanceNameTest/InstanceNameTest.cs

```diff
@@ -84,48 +84,95 @@ public static void ConnectManagedWithInstanceNameTest(bool useMultiSubnetFailove
+            
+            DataTestUtility.ParseDataSource(builder.DataSource, out hostname, out port, out instanceName);
+            
+            Assert.False(string.IsNullOrEmpty(hostname), "Hostname must be included in the data source.");
+            Assert.False(string.IsNullOrEmpty(instanceName), "Instance name must be included in the data source.");
 
-            Assert.True(DataTestUtility.ParseDataSource(builder.DataSource, out string hostname, out _, out string instanceName), "Data source to be parsed must contain a host name and instance name");
+            bool isBrowserRunning = IsBrowserAlive(hostname);
+            Assert.True(isBrowserRunning, "Browser service is not running.");
 
-            bool condition = IsBrowserAlive(hostname) && IsValidInstance(hostname, instanceName);
-            Assert.True(condition, "Browser service is not running or instance name is invalid");
+            bool isInstanceExisting = IsValidInstance(hostname, instanceName);
+            Assert.True(isInstanceExisting, "Instance name is invalid.");
 
-            if (condition)
+            if (isBrowserRunning && isInstanceExisting)
             {
-                using SqlConnection connection = new(builder.ConnectionString);
+                // Create a connection object to ensure SPN info is available via reflection
+                SqlConnection connection = new(builder.ConnectionString);
                 connection.Open();
-                using SqlCommand command = new("SELECT auth_scheme, local_tcp_port from sys.dm_exec_connections where session_id = @@spid", connection);
-                using SqlDataReader reader = command.ExecuteReader();
-                Assert.True(reader.Read(), "Expected to receive one row data");
-                Assert.Equal("KERBEROS", reader.GetString(0));
-                int localTcpPort = reader.GetInt32(1);
-
-                int spnPort = -1;
-                string spnInfo = GetSPNInfo(builder.DataSource, out spnPort);
-
-                // sample output to validate = MSSQLSvc/machine.domain.tld:spnPort"
-                Assert.Contains($"MSSQLSvc/{hostname}", spnInfo);
-                // the local_tcp_port should be the same as the inferred SPN port from instance name
-                Assert.Equal(localTcpPort, spnPort);
+
+                // Get the SPN info using reflection
+                string spnInfo = GetSPNInfo(builder.DataSource);
+
+                // The expected output to validate is supposed to be in the format "MSSQLSvc/machine.domain.tld:spnPort".
+                // So, we want to get the port number from the SPN and ensure it is a valid port number.
+                string[] spnStrs = spnInfo.Split(':');
+                int portInSPN = 0;
+                if (spnStrs.Length > 1)
+                {
+                    int.TryParse(spnStrs[1], out portInSPN);
+                }
+                Assert.True(portInSPN > 0, "The expected SPN must include a valid port number.");
```

Comment:
Assert.Equal now in use as suggested. An SPN pattern validation is added that uses regular expression because a localhost name in the data source gets resolved into a machine name inside the SPN, for example "MSSQLSvc/LAPTOP-XYZ:1433".